Prepare for General Registry registration

hsugawa8651 opened the pull request:
This PR prepares FCCQuad.jl for registration in the Julia General Registry.

# Changes

## Project.toml

- Add [compat] section with julia = "1.6" (LTS version support)

##  Documentation

- Improve docstrings following Julia documentation guidelines
- Add Documenter.jl setup (docs/make.jl, docs/src/index.md)
- Documentation will be deployed to GitHub Pages automatically

## CI/CD

- Add GitHub Actions workflow for CI (.github/workflows/CI.yml)
  - Tests on Julia 1.6, 1.10, and nightly
  - Runs on ubuntu-latest, windows-latest, macOS-latest
- Add GitHub Actions workflow for documentation deployment

## README

- Add badges (CI status, documentation, license)

# Next Steps

After merging, the package owner can register by commenting:
@JuliaRegistrator register

# Related

Closes #1

# Note
CI will start running after this PR is merged, since the workflow files are being added in this PR.

# Note on version number:

The current version is 0.4.16. 
According to https://juliaregistries.github.io/RegistryCI.jl/stable/guidelines/, new package registrations should use a "standard initial version number, e.g. 0.0.1, 0.1.0, or 1.0.0."

Options:
1. Keep v0.4.16 -  If AutoMerge fails due to non-standard initial version, request manual merge in Julia Slack #pkg-registration channel with a link to the 
registry PR and explanation.

Sources:
- https://github.com/JuliaRegistries/General/blob/master/README.md
- https://help.juliahub.com/juliahub/stable/registering/

2. Bump to v1.0.0 - Indicates the package is stable; AutoMerge will pass

@dkm2 What is your preference?

hsugawa8651 committed:
Prepare Project.toml for Julia General Registry registration

- Add [compat] section with julia >= 1.6 and dependency versions
- Move Test and QuadGK from [deps] to [extras] (test-only dependencies)
- Add [targets] section for test dependencies
- Add .DS_Store and Manifest.toml to .gitignore

hsugawa8651 committed:
Improve documentation and add GitHub Actions workflows

- Rewrite docstrings following Julia documentation guidelines
  - Use 4-space indented signatures
  - Add @ref links for cross-references
  - Use imperative mood (Integrate, not Integrates)
- Add docstrings for module FCCQuad, constants, and Jets.Jet
- Add Documenter.jl setup (docs/)
- Add GitHub Actions for CI and documentation deployment
- Add CI and documentation badges to README
- Update .gitignore for docs build artifacts

szabo137 commented on .gitignore:
```diff
@@ -5,3 +5,7 @@
+.DS_Store
+Manifest.toml
+docs/build/
+docs/Manifest.toml
```
I think this is redundant, because any `Manifest.toml` is ignored by line 9.

szabo137 commented on Project.toml:
```diff
@@ -4,9 +4,20 @@ authors = ["David Milovich <david.milovich@welkinsciences.com>"]
+HDF5 = "0.17"
+Scratch = "1"
+SpecialFunctions = "2"
+julia = "1.6"
```
Similar to the test script of the gh actions, I propose to use '1.10' as the baseline, because it is the current LTS. Or is there any reason to support '1.6'?

dkm2 replied:
No reason on my end. I merely haven't yet tested FCCQuad.jl against more recent versions of Julia.

szabo137 reviewed (approved):
Many thanks, very well done! 

I just added some comments, but I think this would make `FCCQuad` capable for being registered. We just need to check, if the CI passes after merge.

hsugawa8651 committed:
Address review comments: bump to v1.0.0 and update Julia compat

- Bump version to 1.0.0 for registry registration
- Update Julia compat from 1.6 to 1.10 (current LTS)
- Update CI workflow to test Julia 1.10 instead of 1.6
- Remove redundant docs/Manifest.toml from .gitignore
